opencti-platform/opencti-front/src/private/components/observations/indicators/IndicatorAddObservablesLines.js

```diff
@@ -71,143 +75,184 @@ export const indicatorMutationRelationDelete = graphql`
   }
 `;
 
-class IndicatorAddObservablesLinesContainer extends Component {
-  constructor(props) {
-    super(props);
-    this.state = { expandedPanels: {} };
-  }
+const IndicatorAddObservablesLinesContainer = (props) => {
+  const { indicator, indicatorObservables, data } = props;
+  const classes = useStyles();
+  const { t } = useFormatter();
+
+  const [commitRelationAdd] = useMutation(indicatorMutationRelationAdd);
+  const [commitRelationDelete] = useMutation(indicatorMutationRelationDelete);
+
+  const enableReferences = useIsEnforceReference('stix-core-relationship');
+  const [expandedPanels, setExpandedPanels] = useState({});
+  const [showForm, setShowForm] = useState(false);
+  const [selected, setSelected] = useState(null);
 
-  toggleStixCyberObservable(stixCyberObservable) {
-    const { indicatorId, indicatorObservables } = this.props;
-    const indicatorObservablesIds = map((n) => n.node.id, indicatorObservables);
-    const alreadyAdded = indicatorObservablesIds.includes(
-      stixCyberObservable.id,
-    );
+  const handleOpenForm = () => {
+    setShowForm(true);
+  };
+  const handleCloseForm = () => {
+    setShowForm(false);
+  };
+
+  const toggleStixCyberObservable = (stixCyberObservable, alreadyAdded) => {
+    const input = {
+      fromId: indicator.id,
+      toId: stixCyberObservable.id,
+      relationship_type: 'based-on',
+    };
+    // Delete
     if (alreadyAdded) {
-      commitMutation({
-        mutation: indicatorMutationRelationDelete,
-        variables: {
-          fromId: indicatorId,
-          toId: stixCyberObservable.id,
-          relationship_type: 'based-on',
-        },
-        updater: (store) => {
-          const conn = ConnectionHandler.getConnection(
-            store.get(indicatorId),
-            'Pagination_observables',
-          );
-          ConnectionHandler.deleteNode(conn, stixCyberObservable.id);
-        },
+      commitRelationDelete({
+        variables: { ...input },
+        updater: (store) => deleteNodeFromEdge(store, 'observables', indicator.id, stixCyberObservable.id, { first: 200 }),
       });
+      // Add with references
+    } else if (enableReferences) {
+      handleOpenForm();
+      setSelected(stixCyberObservable);
+      // Add
     } else {
-      const input = {
-        fromId: indicatorId,
-        toId: stixCyberObservable.id,
-        relationship_type: 'based-on',
-      };
-      commitMutation({
-        mutation: indicatorMutationRelationAdd,
+      commitRelationAdd({
         variables: { input },
       });
     }
-  }
+  };
 
-  handleChangePanel(panelKey, event, expanded) {
-    this.setState({
-      expandedPanels: assoc(panelKey, expanded, this.state.expandedPanels),
+  const createRelation = (values) => {
+    const input = {
+      ...values,
+      fromId: indicator.id,
+      toId: selected.id,
+      relationship_type: 'based-on',
+    };
+    const finalValues = R.pipe(
```

Comment:
Maybe you could try something like:
```suggestion
    const finalValues = {
        confidence: parseInt(String(values.confidence), 10),
        etc etc
}
```

Reply:
You are right, but it will be done next time :)